### ISSUE TITLE
Multiple prefix when uploading multiple files

### ISSUE THREAD

**Issue by vasilvestre** [dustin10/VichUploaderBundle#1385]
### Bug Report

|    Q            |   A
|---------------- | ------
| BC Break        | no
| Bundle version  | 1.19.1
| Symfony version | 6.0.0
| PHP version     | 8.1

#### Summary

The file path in the database contains the prefix multiple times, depending of how many entities I'm uploading at same time.

#### Current behavior

In the database, the URL looks like this for 3 entities that I'm saving at the same time, with the same picture (sorted by id)

local//local//moi-64424393e90d5857016734.png
local//moi-64424393ee2da517474008.png
moi-64424393f001f623398689.png

#### How to reproduce

I create a CollectionType of a form that contains a VichUploader field. When I save the entities, the bug occurs.

#### Expected behavior

The correct behavior would be to have no prefix saved in a database for all entities.

**Comment by garak** [comment]
What namer are you using?

**Comment by garak** [comment]
The entity is the same?

**Comment by vasilvestre** [comment]
Same entity but different instance

**Comment by garak** [comment]
Could you provide additional information? How the uploadable entity is related to the main one?

**Comment by vasilvestre** [comment]
There's no relationship, it's an unmapped form. It only creates collection of my entity type which contains a VichFileType.

**Comment by garak** [comment]
I'd love to have so much free time to provide personal support, but unfortunately it's not the case.
Also, this is probably not a bug but a misuse of the bundle.
If you can provide the offending code, I can try to help you here

**Comment by vasilvestre** [comment]
> I'd love to have so much free time to provide personal support, but unfortunately it's not the case.
> Also, this is probably not a bug but a misuse of the bundle.
> If you can provide the offending code, I can try to help you here

So, I can now safely say that the issue was due to misuse of event listener.
EasyAdmin seems to do not call every listener, I had to refresh the entity using doctrine so it worked.
Another problem came from this context :
Entity A holds many entities (B) that are VichUploadable. When I persist entity A, I remove it but keep entities B. The doctrine events are not fired despite entities B being flushed. I guess it's also due to EasyAdmin.

Sorry for the inconvenience
